app/src/main/java/net/measurementlab/ndt7/android/MainActivity.kt

```diff
@@ -1,32 +1,85 @@
 package net.measurementlab.ndt7.android
 
+import android.app.Activity
 import android.os.Handler
 import android.os.Looper
-import android.support.v7.app.AppCompatActivity
 import android.os.Bundle
 import android.util.Log
+import android.widget.TextView
 import android.widget.Toast
+import butterknife.BindView
+import butterknife.ButterKnife
+import java.net.URL
+import com.google.gson.Gson
+import net.measurementlab.ndt7.android.model.Server
+import java.util.*
+import kotlin.math.pow
+
 
 private const val TAG = "MainActivity"
+private const val LOCATESERVER = "https://locate.measurementlab.net/ndt7"
+
+class MainActivity : Activity() {
+    @BindView(R.id.server)
+    lateinit var tv_server: TextView
+
+    @BindView(R.id.downloadspeed)
+    lateinit var tv_downloadspeed: TextView
+
+    @BindView(R.id.maxbw)
+    lateinit var tv_maxbw: TextView
+
+    @BindView(R.id.minrtt)
+    lateinit var tv_minrtt: TextView
+
+    @BindView(R.id.smoothedrtt)
+    lateinit var tv_smoothedrtt: TextView
+
+    @BindView(R.id.rttvariance)
+    lateinit var tv_rttvariance: TextView
+
+    @BindView(R.id.uploadspeed)
+    lateinit var tv_uploadspeed: TextView
+
+    val settings = Settings()
 
-class MainActivity : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-        val settings = Settings()
-        settings.hostname = "35.235.104.27"
+        ButterKnife.bind(this)
+
+        val locateUrl = URL(LOCATESERVER)
+        val ndt7ServerSearch = Ndt7ServerSearch(locateUrl)
+        val rawJsonServer = Gson().fromJson(ndt7ServerSearch.server, Server::class.java)
+
+
+        val fqdn    = rawJsonServer.fqdn.replace("\"","")
+        val country = rawJsonServer.country.replace("\"","")
+        val ipv4    = rawJsonServer.ip[0].replace("\"","")
+        val site    = rawJsonServer.site.replace("\"","")
+        val city    = rawJsonServer.city.replace("\"","")
+
+        Log.d(TAG, "SERVER FOUND:$fqdn")
+
+        tv_server.text = String.format(Locale.ENGLISH, "IPv4: %s\nCity: %s\nCountry: %s\n" +
+                "Site: %s",ipv4,city,country,site)
+
+        settings.hostname = fqdn
         settings.port = 443
         settings.skipTlsCertificateVerification = true
-        val client = MyClient(settings)
 
-        if (!client.runDownload()) {
+
+
+        val clientKt = MyClientKt(settings)
+
+        if (!clientKt.runDownload())
             Toast.makeText(this, "runDownload failed", Toast.LENGTH_SHORT).show()
-        }
+
     }
 
-    private inner class MyClient constructor(settings: Settings) : Client(settings) {
+    inner class MyClientJ constructor(settings: Settings) : ClientUpload(settings) {
```

Comment:
Is the `J` here used to indicate that this is `Java`?

Reply:
Yes, it is.